avocado_vt/loader.py

```diff
@@ -79,6 +79,28 @@ def arch_listing(config, guest_name_parser=None):
     LOG.debug("")
 
 
+def get_vt_test_tags(vt_params):
+    """
+    Returns the test categories based on a `tags` vt param
+
+    :rtype: dict
+    """
+    tags = vt_params.get('tags', '').strip()
+    result = {}
+    for tag in tags.split(','):
```

Comment:
How about defining the tags in a more Cartesian config friendly manner through intervals? E.g.
```
tags = A B C
```
with `vt_params.objects('tags')` instead?

Reply:
That's the first idea I got, but then I changed to the current one in order to strictly respect the tags syntax defined by Avocado, because VT is only responsible for providing the data of `tags` and fully rely on the tags implementation given by Avocado [1] [2] [3]. 

Btw, by the current tags implementation of Avocado, IMO, it should provide a function to plugins for parsing tags strings:
```python
from avocado.core.tags import parse_tags

def get_vt_test_tags(vt_params):
    tags = vt_params.get('tags', '').strip()
    return parse_tags(tags)
```
otherwise each plugin have to implement the same thing again and again. Or else, Avocado should allow plugins to redefine the tags syntax and give the implementation of the tags interpreter (include both parser and resolver).

Now back to the original topic, using `' '` as the delimiter is something not logic proper but harmless in the current stage, since `' '` is not a valid character for the tag identifier [4], so it's fine to adopt your suggestion. What do you think?

References:
[1] https://github.com/avocado-framework/avocado/blob/master/avocado/core/tags.py
[2] https://github.com/avocado-framework/avocado/blob/4a6641ab0b7c3f0bb1c568d2929d3955ceda3aba/avocado/core/suite.py#L93-L122
[3] https://github.com/avocado-framework/avocado/blob/4a6641ab0b7c3f0bb1c568d2929d3955ceda3aba/avocado/core/safeloader.py#L268-L290
[4] https://github.com/avocado-framework/avocado/blob/4a6641ab0b7c3f0bb1c568d2929d3955ceda3aba/avocado/core/safeloader.py#L232

Reply:
I think that it is best if we use VT specific syntax (`tags = A B C`) which is also more compatible and readable together with our Cartesian configuration, then satisfy the Avocado requirements right before passing the tags further to it. I think in this way we get the best from both worlds.

Reply:
ok, I'll update it per your suggestion and leave a comment to explain the issue I mentioned. Btw, we can't use `objects()`, it complaints as:
```
$ avocado list --filter-by-tags=fast
Test discovery plugin <avocado_vt.loader.VirtTestLoader object at 0x7f8a4676f490> failed: 'dict' object has no attribute 'objects'
```
will replace with `get().split()`, thanks.

Reply:
Yes good point, I guess it doesn't receive our widely used `Params` object here. How is `vt_params` parsed? Aren't the tags obtained from the Cartesian config?

Reply:
The `Params` wrapper is provided by `virttest.utils_params`, `loader.py` doesn't import that therefore it can't take the benefit.

Reply:
Yes, I was surprised to see that Cartesian parser's `get_dicts()` doesn't in fact return a `Params` object, something I would find much more reasonable considering we use the finally parsed test parameters as its ubiquitous instance. :eyes: